### PR TITLE
erk and knives

### PR DESCRIPTION
**Pull request by Kamejeir**
#### Summary
Features "Various knives now using the variant system"

#### Purpose of change

@I-am-Erk wanted this

#### Describe the solution

Move knives_kitchen.json contents to knives.json

Make the kitchen knives into variants as determined below

Changed a mission needing a knife

Changed around tests to check for the base knives instead of the variants

migrate the knives to their respective variants

Made some mods that need the knives now use any of the knives in their category (sorry crazycata, your bread sword is now no longer just a bread sword. rather chef- and carving sword too)

#### Testing

Small kitchen knife: Loads no fault, spawns in with separate variants, the items get their own ASCII art

further testing: no fault when loading or throughout testing spawning/recipes

found one of each knife and made this beauty
image

all knives disassemble successfully

a practice recipe could successfully utilize a knife

I'm not sure how to see about checking the mod changes over just seeing that the unit tests passed

### CODE REVIEW

File: data/json/itemgroups/Food/food.json [hunk 1]
```diff
@@ -851,7 +851,7 @@
       { "prob": 45, "group": "cheese_bag_plastic" },
       { "item": "mozzarella_cheese", "prob": 30 },
       { "item": "ceramic_plate", "prob": 30 },
-      { "item": "knife_steak", "prob": 45 },
+      { "item": "knife_small", "prob": 45 },
```
Comment (I-am-Erk) [inline data/json/itemgroups/Food/food.json:854]:
```suggestion
      { "item": "knife_small", "variant": "knife_steak", "prob": 45 },
```
You'll want to make sure most of these have the same chances of spawning a steak or paring knife appropriately, a paring knife isn't usually something you'd see in a dining set for example.

**Comment by I-am-Erk** [comment]
You can also keep them in the current knife file, it really doesn't matter much

**Comment by Kamejeir** [comment]
Done with the small kitchen knives. I don't know which itemgroups would deserve the extra knives such as the "tomato knife". I mean obviously yes wherever there are all variants (like a kitchen), but scanning for exceptions is the tricky part. Also I'm not too keen on adding descriptions of the extra knives, am not a knife enthusiast.

**Comment by Kantonine** [comment]
Also, quick thing to add, both cleavers should have the knife proficiencies

**Comment by I-am-Erk** [comment]
This is looking good so far. I would not bother with adding new knife variants in this PR, that's just something we can think about later.

**Comment by Kamejeir** [comment]
Here includes all variants of the large knife. Is this the best way to spawn these three? Anything like taking out the variant definitions might not be future-proof, though in case more large knives are added...
https://github.com/CleverRaven/Cataclysm-DDA/blob/3b3b739429df5f34fc4b329976a6bb876fd8ea50/data/json/itemgroups/collections_domestic.json#L101-L103
This becomes rather silly. One can potentially substitute in a bread knife for use in butchery, which hardly seems appropriate.
"Kit" items are so last year anyway.
https://github.com/CleverRaven/Cataclysm-DDA/blob/3b3b739429df5f34fc4b329976a6bb876fd8ea50/data/json/recipes/tools/tool.json#L231

**Comment by I-am-Erk** [comment]
You could keep the bread knife as a separate item if it makes you more comfortable. I don't personally think it's all that significant a problem

**Comment by Kamejeir** [comment]
My only concern is this. It's not letting me use variants, so a new huge knife variant would make it not necessarily be giving this NPC a butcher knife. Maybe the description of the mission can be spiced up as more generic, assuming future huge knives all have function similar to a butcher knife?
https://github.com/CleverRaven/Cataclysm-DDA/blob/de4ec39f57b1be19b9975e42971671b609c37bf9/data/json/npcs/godco/godco_missions.json#L60

**Comment by Kamejeir** [comment]
@I-am-Erk I'll go with this. I'll test it now and once you're good with how the mission turned out, I'll put it as ready for review.

**Comment by Kamejeir** [comment]
oh no, tests

**Comment by Kamejeir** [comment]
I'm angry
image

**Comment by Kamejeir** [comment]
I don't feel like doing this.

**Comment by Kamejeir** [comment]
image
Could probably work backwards from this, but I don't know anything about spells. Or even if it is limited to Magiclysm anymore.

**Comment by I-am-Erk** [comment]
Ah yeah. The magiclysm conflicts and resulting explosion are a good case study of why it might have been better to do just steak/paring knives, then just bigger knives, in separate PRs. It looks like you have it mostly sorted now though, if it passes these tests I'll merge it in a bit.

**Comment by Kamejeir** [comment]
Did my best to make it less bad for the mergers.

**Comment by Kamejeir** [comment]
Commit readability done. Time to wait for tests. If nothing comes up, I think it's good to go?

**Comment by Kamejeir** [comment]
o-oh no. I will have to mess with migration, won't I?

**Comment by Kamejeir** [comment]
Self-reviewing is good, even if the only thing I managed to catch was this instance of "knife_butcher" right before it was overwritten.
largeknifecommit

Won't matter much, now that I have to figure out what to do regarding to prevent knives evaporating from every loaded game following this PR.

**Comment by Kamejeir** [comment]
fixed handling obsoletion. I start to understand that the commit log needs to both serve as a changelog while also being readable. good experience for the future

**Comment by Maleclypse** [comment]
Please rebase to resolve conflicts

**Pull request merged**